apps/website/src/middleware.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  type NextFetchEvent,
+  type NextRequest,
+  NextResponse,
+} from "next/server";
+import { env } from "@/env/index.mjs";
+
+export async function middleware(req: NextRequest, event: NextFetchEvent) {
+  const res = await fetch(env.NEXT_PUBLIC_BASE_URL + "/api/short-links", {
+    method: "GET",
+  });
+  const shortLinks = await res.json();
+
+  for (const link of shortLinks) {
+    if (req.nextUrl.pathname.startsWith("/l/" + link.slug)) {
```

Comment:
Does it make sense to use `startsWith` here? Should the slug not be a full match. If i understand correctly a slug ' twitch' would always win over `twitch-giveaway`, `twitchsub` etc. if it was added first?

I think next already normalizes trailing slashes, so that _should_ not be a problem.